views/RepoList.tsx

```diff
@@ -56,14 +56,17 @@ const RepoList = (props: { repo_list: RepoIdentifier[] }) => {
 export const getRepoList = async (conn: Pool, session: Session) => {
 	const allRepos = await getUserRepositories(session);
 	const allReposFormattedAsTuples = allRepos.map(repo => `(${convert(repo.repo_provider)}, ${convert(repo.repo_owner)}, ${convert(repo.repo_name)})`).join(',');
-	const repo_list_q = `SELECT * FROM repos 
-	WHERE (repo_provider, repo_owner, repo_name) IN (${allReposFormattedAsTuples})`;
-	const result = await conn.query(repo_list_q);
-	return result.rows.map(row => ({
-		repo_provider: row.repo_provider,
-		repo_owner: row.repo_owner,
-		repo_name: row.repo_name
-	}));
+	const repo_list_q = `SELECT *
+		FROM repos 
+		WHERE (repo_provider, repo_owner, repo_name) IN (${allReposFormattedAsTuples})`;
+	const result: { rows: DbRepo[] } = await conn.query(repo_list_q).catch(err => {
+		console.error(`[RepoList] Error in getting repository-list from the database`, err);
+		throw Error(err); // FIXME: handle this more elegantly
+	});
+	return result.rows.map(repo => {
+		const { created_at, ...other } = repo;
+		return { created_at: created_at.toDateString(), ...other }
```

Comment:
What does this syntax mean? What's other? Asking for my personal knowledge

Reply:
this is the `deconstruction` syntax. If repo is an object with multiple fields, one of which is `created_at`, then this syntax will put the value of the `created_at` field in the variable named `created_at` and all the other fields (and only the other fields) will be in the object named `other`.